File: output/GPT4o-Mini/xCodeEval_4000/C++/Java/6269f2e0b6a350817fb7440d60199d62.java
import java.util.Scanner;

class Main {
    static class ModInt {
        private static final int MOD = 1000000007;
        private long value;

        public ModInt(long value) {
            this.value = value % MOD;
            if (this.value < 0) this.value += MOD;
        }

        public long val() {
            return value;
        }

        public ModInt add(ModInt other) {
            return new ModInt(this.value + other.value);
        }

        public ModInt multiply(ModInt other) {

File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/Python/6269f2e0b6a350817fb7440d60199d62.py
class ModInt:
    def __init__(self, value, mod=1000000007):
        self.mod = mod
        self.value = value % mod

    def __add__(self, other):
        return ModInt(self.value + other.value, self.mod)

    def __mul__(self, other):
        return ModInt(self.value * other.value, self.mod)

    def val(self):
        return self.value


def main():
    import sys
    input = sys.stdin.read
    data = input().split()
    
    q = int(data[0])
    index = 1
    results = []
    
    for _ in range(q):
        n = int(data[index])
        index += 1
        M = n // 4 + 1
        m = ModInt(M)
        
        if n % 4 <= 1:
            results.append((m * m).val())
        else:
            results.append((m * (m + ModInt(1))).val())
    
    print("\n".join(map(str, results)))


if __name__ == "__main__":
    main()
